sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/CatalogPlugin.java

```diff
@@ -74,4 +74,12 @@ public interface CatalogPlugin {
   default String[] defaultNamespace() {
     return new String[0];
   }
+
+  /**
+   *
+   * @return true if catalog supports multi-part namespaces
+   */
+  default boolean supportsMultiPartNamespace() {
+    return false;
```

Comment:
Hm, is this a breaking change given that existing catalogs that actually support multi-part namespaces may be unaware of this method?

Reply:
Makes sense, changed to not populate 'namespace' unless there's multi-part.  There's not a ton of value in having an empty array just because the catalog supports it, for a non-multi-part-namespace procedure.